### PR TITLE
Make the main menu open from the hamburger button too.

### LINKED ISSUE #305: The main menu isn't available on Windows
Either because it is OSX only or because we've hidden the titlebar or something but the menu that we're making changes to isn't visible to windows users. We must fix that or remove it and use something else that Windows users can see.

### PR DESCRIPTION
**Pull request by Mossop**
We cache the bookmarks data so we can rebuild the menu at any time and some
bits are moved around to allow for building menus for minimal OSX, regular
application and the hamburger menu. The hamburger menu has a slightly different
template allowing important options to be at the top level.

Fixes #305.

### CODE REVIEW

File: app/main/menu/index.js [hunk 1]
```diff
@@ -11,58 +11,60 @@ specific language governing permissions and limitations under the License.
...
+    }
+
+    return itemTemplate;
+  };
+
+  return template.map(buildItem);
+}
+
+function buildMenu(template, data) {
+  return currentMenu = electron.Menu.buildFromTemplate(buildTemplate(template, data));
```
Comment (bgrins) [inline app/main/menu/index.js]: Nit: please split this up into two lines

**Comment by bgrins** [comment]
With this applied, the "File" menu is missing on the OSX app menu bar

**Comment by bgrins** [comment]
screenshot 2016-05-19 09 30 45

**Comment by Mossop** [comment]
@bgrins That should fix it but you should probably check

File: app/ui/browser/lib/webview-controller.js [hunk 2]
```diff
@@ -71,6 +71,16 @@ class WebViewController extends EventEmitter {
     this.emit('navigate-refresh', pageId);
     this.emit(`navigate-refresh:${pageId}`, pageId);
   }
+
+  capturePage(pageId) {
+    console.log(3);
```
Comment (bgrins) [inline app/ui/browser/lib/webview-controller.js]: Leftover console.log()

**Comment by bgrins** [comment]
Nice

**Pull request merged**